spark/src/test/scala/org/apache/mahout/sparkbindings/test/DistributedSparkSuite.scala

```diff
@@ -45,6 +45,7 @@ trait DistributedSparkSuite extends DistributedMahoutSuite with LoggerConfigurat
           .set("spark.akka.frameSize", "30")
           .set("spark.default.parallelism", "10")
           .set("spark.executor.memory", "2G")
+          .set("spark.kryo.registrationRequired", "true")
```

Comment:
This is not needed, this is why it is failing. we can enable that to see what else is left, but we don't have to patch every class out there that is used in tests. I think the classes that still have that have something to do with IndexedDataSet which is (in my view) not part of algebra engine, so we can ignore the rest.